fable/include/fable/schema/factory.hpp

```diff
@@ -58,18 +57,56 @@ class FactoryBase : public Base<CRTP> {
  public:  // Types
   using Type = T;
   using MakeFunc = std::function<T(const Conf& c)>;
+
+  /**
+   * TypeFactory is basically a pair with names for better readability.
+   */
   struct TypeFactory {
-    TypeFactory(const Box& s, MakeFunc f) : schema(s), func(f) { schema.reset_ptr(); }
+    TypeFactory(Box s, MakeFunc f) : schema(std::move(s)), func(std::move(f)) {
+      schema.reset_ptr();
+    }
 
-    Box schema;
-    MakeFunc func;
+    Box schema;     // NOLINT
+    MakeFunc func;  // NOLINT
   };
 
   using TransformFunc = std::function<Box(Struct&&)>;
   using FactoryMap = std::map<std::string, TypeFactory>;
   using FactoryPairList = std::initializer_list<std::pair<std::string, TypeFactory>>;
 
  public:  // Constructors
+  ~FactoryBase() noexcept override = default;
+
+ protected:
+  FactoryBase(const FactoryBase& other)
+      : Base<CRTP>(other)
+      , transform_func_(other.transform_func_)
+      , available_(other.available_)
+      , factory_key_(other.factory_key_)
+      , args_key_(other.args_key_)
+      , args_subset_(other.args_subset_) {
+    reset_schema();
+  }
+
+  FactoryBase& operator=(const FactoryBase& other) {
+    if (&other == this) {
+      return *this;
+    }
```

Comment:
It may have. I can't remember. But this check is important for the edge case and is used in official implementations.

See: https://en.cppreference.com/w/cpp/language/copy_assignment